src/MKL.jl

```diff
@@ -1,9 +1,53 @@
 module MKL
 
-using MKL_jll
+using Preferences
+using Libdl
 
-JULIA_VER_NEEDED = v"1.7.0-DEV.641"
-VERSION > JULIA_VER_NEEDED && using LinearAlgebra
+# Choose an MKL provider; taking an explicit preference as the first choice,
+# but if nothing is set as a preference, fall back to an environment variable,
+# and if that is not given, fall back to the default choice of `MKL_jll`.
+const mkl_provider = lowercase(something(
+    @load_preference("mkl_provider", nothing),
+    get(ENV, "JULIA_MKL_PROVIDER", nothing),
```

Comment:
It's worth noting that the value of `JULIA_MKL_PROVIDER` is only read at `.ji` compile time.